src/core/core.test.ts

```diff
@@ -15,7 +15,7 @@ describe('core', () => {
 	});
 
 	test('creates a fixture', () => {
-		const core = new Core().register(defaultGenerators);
+		const core = new Core().register(fixtureGenerators);
```

Comment:
I like the restructure, but here I am again bringing up the name `Core` 😅
I wonder if we can rename it to something else.
Core makes sense for the internals, but I don't know if it's a good name to export.

`Core` doesn't make the intent clear if a user would import it (and maybe there are multiple libraries that export a `Core`).

The name I bring up again is `Fixture`.
And I would like to hear your thoughts on the following example(s):

```ts
// manual
const fixture = new Fixture().register(fixtureGenerators);
const value = fixture.generate(zodSchema) 

// predefined
// this is different than the current API because 
// createFixture returns the preconfigured fixture including the default generators
// to generate a value, generate needs to be invoked
const fixture = createFixture() // implementation would be new Fixture().register(defaultGenerators);
const value = fixture.generate(zodSchema) // or fixture.create

// predefined + extended
const fixture = createFixture() 
fixture.extend(customGenerator)
const value = fixture.generate(zodSchema)  // or fixture.create


// serialize (also preconfigured)
const fixture = createSerializableFixture() 
const value = fixture.generate(zodSchema)  // or fixture.create
```

This would also make it possible to reuse a fixture across multiple tests with the same seed, which isn't possible with `createFixture` without updating all references.


I promise that it's the last time that I bring this up 😅

Reply:
No no, keep bringing things up! These are good thought exercises and valid points. You're definitely on to something with the ability to set a global seed value and core IS very generic and doesn't describe what the class is actually doing. Let me contemplate these issues and explore alternatives.

Reply:
I've been thinking about what `Core` is and what `Core` does. It's essentially a "transpiler" that takes a Zod schema and transforms it into another object with a similar structure.

```ts
// manual
const fixture = new Transpiler({ seed: 1234 }).extend(fixtureGenerators);
const value = fixture.from(zodSchema) // same as fixture.generate

// we can now extend the Transpiler class
export class Fixture extends Transpiler {
	generators = fixtureGenerators;

	// explicitly define the return type
	generate<TSchema extends z.ZodTypeAny>(
		schema: TSchema,
		context?: Context
	): z.infer<TSchema> {
		return super.generate(schema, context);
	}
}

// predefined
// this is different than the current API because 
const fixture = new Fixture({ seed: 1234 })
const value = fixture.from(zodSchema) // same as fixture.generate

// predefined + extended
const fixture =  new Fixture({ seed: 1234 })
fixture.extend(customGenerator)
const value = fixture.from(zodSchema)  // same as fixture.generate

// serializer (also preconfigured by extending Transpiler)
const serializer = new Serializer({ seed: 1234 })
const value = serializer.from(zodSchema)  // same as serializer.generate
```

Reply:
Or possibly Translate or Translator?

Reply:
I like `Transformer` the most